### PR TITLE
add some unit tests for pkg/api

### PR DESCRIPTION
**Pull request by qiuming520**
Signed-off-by: qiuming520 <qiuming_yewu@cmss.chinamobile.com>

### What this PR does / why we need it:
add some unit tests for pkg/api
### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by benjaminhuo** [comment]
@junotx @wanjunlei Would you help to review this?

File: pkg/api/logging/v1alpha2/types_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	request := restful.NewRequest(req)
+	_, err = ParseQueryParameter(request)
+	assert.NoError(t, err)
+
+	// interval operation
+	queryParamInterval := "operation=interval&start_time=1136214245&end_time=1136214245&from=0"
```
Comment (wanjunlei) [inline pkg/api/logging/v1alpha2/types_test.go:25]: The operation should be one of `query `, `statistics `, `histogram `, and `export`, the `interval` should define by [interval](https://github.com/kubesphere/kubesphere/blob/master/pkg/kapis/tenant/v1alpha2/register.go#L262) parameter.

File: pkg/api/logging/v1alpha2/types_test.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	reqInterval, err := http.NewRequest("GET", fmt.Sprintf("http://localhost/tenant.kubesphere.io/v2alpha1/logs?%s", queryParamInterval), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	requestInterval := restful.NewRequest(reqInterval)
+	_, err = ParseQueryParameter(requestInterval)
```
Comment (wanjunlei) [inline pkg/api/logging/v1alpha2/types_test.go:32]: Maybe we should check the return value of `ParseQueryParameter`, null error does not mean the parameter parsing is as expected.

File: pkg/api/logging/v1alpha2/types_test.go (hunk 1 above)
Comment (qiuming520) [reply pkg/api/logging/v1alpha2/types_test.go]: this is my mistake! thanks ! I have fix it . @wanjunlei

File: pkg/api/logging/v1alpha2/types_test.go (hunk 2 above)
Comment (qiuming520) [reply pkg/api/logging/v1alpha2/types_test.go]: yeah! I have added assertion of the result

File: pkg/api/logging/v1alpha2/types_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	queryParam := "start_time=1136214245&end_time=1136214245&from=0"
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://localhost/tenant.kubesphere.io/v2alpha1/logs?%s", queryParam), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	request := restful.NewRequest(req)
+	ret, err := ParseQueryParameter(request)
+	assert.NoError(t, err)
+	assert.Equal(t, ret.Operation, OperationQuery)
```
Comment (wanjunlei) [inline pkg/api/logging/v1alpha2/types_test.go:23]: I think unit tests should cover all the code as much as possible, so all fields in `Parameter` need to be checked.

Comment (qiuming520) [reply pkg/api/logging/v1alpha2/types_test.go]: As you suggested, I have added all fields in Parameter need to be checked . @wanjunlei

File: pkg/api/auditing/v1alpha1/types_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+func TestParseQueryParameter(t *testing.T) {
+	queryParam := "operation=query&workspace_filter=my-ws,demo-ws&workspace_search=my,demo&objectref_namespace_filter=my-ns,my-test&objectref_namespace_search=my&objectref_name_filter=my-ref,demo-ref&objectref_name_search=ref&source_ip_search=192.168.&user_filter=user1&user_search=my,demo&group_search=my,demo&start_time=1136214245&end_time=1136214245&from=0&sort=desc"
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://localhost/tenant.kubesphere.io/v2alpha1/auditing/events?%s", queryParam), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	request := restful.NewRequest(req)
```
Comment (wanjunlei) [inline pkg/api/auditing/v1alpha1/types_test.go:20]: We can make the code more readable like this

```
	expected := Query{
		Operation:                  "query",
		WorkspaceFilter:            "my-ws,demo-ws",
		WorkspaceSearch:            "my,demo",
		ObjectRefNamespaceFilter:   "my-ns,my-test",
		ObjectRefNamespaceSearch:   "my",
		ObjectRefNameFilter:        "my-ref,demo-ref",
		ObjectRefNameSearch:        "ref",
		UserFilter:                 "user1",
		UserSearch:                 "my,demo",
		GroupSearch:                "my,demo",
		SourceIpSearch:             "192.168.",
		StartTime:                  time.Unix(1136214245, 0),
		EndTime:                    time.Unix(1136214245, 0),
		Interval:                   "15m",
		Sort:                       "desc",
		From:                       0,
		Size:                       10,
	}
	
	request := restful.NewRequest(req)
	actual, err := ParseQueryParameter(request)
	assert.NoError(t, err)
	assert.Equal(t, expected, actual)
```

**Review by wanjunlei — APPROVED** [review]

**Comment by benjaminhuo** [comment]
@qiuming520 Thank you for the contribution!

**Pull request merged**